examples/hackernews/back-end/reactive_service/src/hackernews.service.ts

```diff
@@ -26,38 +25,28 @@ type Upvote = {
 
 type Upvoted = Post & { upvotes: number; author: User };
 
-export default class HackerNewsService implements SkipService {
-  initialData: {
-    posts: Entry<string, Post>[];
-    users: Entry<string, User>[];
-    upvotes: Entry<string, Upvote>[];
+export function serviceWithInitialData(
+  posts: Entry<string, Post>[],
+  users: Entry<string, User>[],
+  upvotes: Entry<string, Upvote>[],
+): SkipService {
+  return {
+    initialData: { posts, users, upvotes },
+    resources: { posts: PostsResource },
+    createGraph: (inputCollections: {
+      posts: EagerCollection<number, Post>;
+      users: EagerCollection<number, User>;
+      upvotes: EagerCollection<number, Upvote>;
+    }) => {
```

Comment:
Something like https://github.com/SkipLabs/skip/pull/501/commits/18a4a7eaa7d0dfc57fbe78a063bc3d62be0aca65 ?

That gives me the idea that we can type-parameterize services more generally, using these `Record<string, EagerCollection<...>>` things as bounds instead of directly in types... will give that a shot, see how it looks.